src/main/cpp/locationinfo.cpp

```diff
@@ -102,6 +120,14 @@ const char* LocationInfo::getFileName() const
 	return fileName;
 }
 
+/**
+ *   Return the short file name of the caller.
+ *   @returns file name, may be null.
+ */
+const std::string LocationInfo::getShortFileName() const{
```

Comment:
Wouldn't `const std::string&` make more sense here? This way one doesn't rely on compilers optimizing copies away and returning a reference seems always safe to me. The other getter even returns `const char*`, though, I prefer references if objects are available anyway already.

Reply:
This should probably be a `const char*` for consistency.  There's a probably slightly more efficient way to do this as well without relying on `std::string`

Reply:
Sad